packages/nuxt/src/app/composables/router.ts

```diff
@@ -259,3 +262,17 @@ export const setPageLayout = (layout: unknown extends PageMeta['layout'] ? strin
 export function resolveRouteObject (to: Exclude<RouteLocationRaw, string>) {
   return withQuery(to.path || '', to.query || {}) + (to.hash || '')
 }
+
+/**
+ * @internal
+ */
+export function encodeURL (location: string, isExternalHost = false) {
+  const url = new URL(location, 'http://localhost')
+  if (!isExternalHost) {
+    return url.toString().replace(/^http:\/\/localhost\//, '/')
```

Comment:
instead of using `replace`, is it more performant to construct the string with URL properties ?

Reply:
Like:
```suggestion
    return url.pathname + url.search + url.hash
```

Reply:
like that yes :+1: